pkg/registry/thirdpartyresourcedata/codec.go

```diff
@@ -263,6 +263,12 @@ func (t *thirdPartyResourceDataCreator) New(version, kind string) (out runtime.O
 	case "ThirdPartyResourceDataList":
 		return &experimental.ThirdPartyResourceDataList{}, nil
 	default:
-		return t.delegate.New(latest.GroupOrDie("experimental").Version, kind)
+		for _, g := range latest.AllGroups {
```

Comment:
The input argument `version` will be groupVersion after #14156, so no need to plumb group to this interface.

However, even after #14156, this particular thirdPartyResourceDataCreator.New() function won't work, because the passed in groupVersion is the one of thirdPartyStorage, which has nothing to do with the groupVersion of the kind.